### ISSUE TITLE
AD0001 after upgrading from 7.4 to 7.5

### ISSUE THREAD

**Issue by RalfKoban** [SonarSource/sonar-csharp#1831]
### Description
After I upgraded from 7.4 to 7.5, following errors have been reported by Visual Studio and the compilation of my projects failed.

```
error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.EmptyNullableValueAccess' threw an exception of type 'System.NotSupportedException' with message 'ConstantPattern'.
error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.EmptyCollectionsShouldNotBeEnumerated' threw an exception of type 'System.NotSupportedException' with message 'ConstantPattern'.
error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.InvalidCastToInterface' threw an exception of type 'System.NotSupportedException' with message 'ConstantPattern'.
error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce' threw an exception of type 'System.NotSupportedException' with message 'ConstantPattern'.
error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.ConditionEvaluatesToConstant' threw an exception of type 'System.NotSupportedException' with message 'ConstantPattern'.
error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.NullPointerDereference' threw an exception of type 'System.NotSupportedException' with message 'ConstantPattern'.
```

### Related information

* SonarC# Version 7.5
* Visual Studio Version 15.8
* C# Version 7.3

**Comment by valhristov** [comment]
Hi @RalfKoban, it seems that when we enabled the support for C# 7 features we did not cover all possible usages of the ConstantPattern syntax. For the time being I would suggest adding the `/nowarn:AD0001` switch to your build so that it passes.

To my regret it would be difficult to send me a reproducer since we don't print the name of the file where the error is happening... Regardless, I'll find a way to reproduce and fix the problem.
